include/alpaka/block/shared/dyn/BlockSharedMemDynUniformCudaHipBuiltIn.hpp

```diff
@@ -4,13 +4,13 @@
 
 #pragma once
 
-#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+#include "alpaka/block/shared/dyn/Traits.hpp"
+#include "alpaka/core/BoostPredef.hpp"
+#include "alpaka/core/Concepts.hpp"
 
-#    include <alpaka/block/shared/dyn/Traits.hpp>
-#    include <alpaka/core/BoostPredef.hpp>
-#    include <alpaka/core/Concepts.hpp>
+#include <type_traits>
```

Comment:
General question: Do we really want to include the standard library headers if the rest of the file is not enabled?

Reply:
Yes, this is necessary for the amalgamation tool to work, because it will remove all includes after the first one. If the first include is only conditionally included and the condition is not always true (which happens), the include never happens in the amalgamated header at all, leading to missing declarations.